### PR TITLE
feat(`forge`): support for `console.log` on Invariant handlers

### LINKED ISSUE #4951: feat(forge): enable console logging in invariant handlers
### Component

Forge

### Describe the feature you would like

It would be useful to be able to use `console2.log` in invariant handlers, e.g.

```solidity
function MyInvariantTest is StdInvariant {
    Handler internal handler = new Handler();
    function setUp() public virtual {
        targetContract(address(handler));
    }
}

contract Handler {
    function doSomething() external {
        console2.log("Hello World");
    }
}
```

This code does not console log "Hello World" at the moment.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Closes #4951 

## Solution

Adds support for console logs on invariant handlers, by persisting the logs and traces for the last call.

To avoid any performance regressions we only move values and only copy where absolutely necessary (filling each invariant with the corresponding logs at the end), and only persist the call values at the end of each run campaign, as they would just be overwriting values persisted at the stack otherwise.

### CODE REVIEW

**Comment by Evalir** [comment]
hey hey @mds1 mind sanity testing this too? this is a highly requested feature so wanna make sure all is good ^^

**Comment by mds1** [comment]
For sure, again tagging @lucas-manuel and @PaulRBerg for testing assistance too 😁

File: evm/src/fuzz/invariant/executor.rs [hunk 1]
```diff
@@ -552,8 +564,12 @@ fn collect_data(
     }
 }
 
+type RichInvariantResults =
+    (bool, Option<BTreeMap<String, RawCallResult>>, Vec<Log>, Option<CallTraceArena>);
+
```
Comment (mattsse) [inline evm/src/fuzz/invariant/executor.rs:569]: I'd prefer a helper struct for this, makes changing it easier

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit

**Comment by Evalir** [comment]
cc @mattsse this should be good for another review round—functionality is sound

**Review by mattsse — APPROVED** [review]

**Pull request merged**